### PR TITLE
[Workspace] Set the default color to first color from the palette

### PR DESCRIPTION
**Pull request by Kapian1234**
### Description

Set the default color to first color from the palette in create workspace page

### Issues Resolved

https://github.com/opensearch-project/OpenSearch-Dashboards/issues/7635

## Screenshot
Screenshot 2024-08-06 at 13 56 02

Screenshot 2024-08-06 at 13 55 23

## Changelog

- feat: [Workspace] Set default color for workspace create form
### Check List

- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [ ] Commits are signed per the DCO using --signoff

### CODE REVIEW

File: src/plugins/workspace/public/components/workspace_creator/workspace_creator.tsx [hunk 1]
```diff
@@ -107,6 +113,7 @@ export const WorkspaceCreator = (props: WorkspaceCreatorProps) => {
               permissionEnabled={isPermissionEnabled}
               dataSourceManagement={dataSourceManagement}
               availableUseCases={availableUseCases}
+              defaultValues={{ color: euiPaletteColorBlind()[0] }}
```
Comment (wanglam) [inline src/plugins/workspace/public/components/workspace_creator/workspace_creator.tsx:116]: Can we separate this variable to a constant?

File: src/plugins/workspace/public/components/workspace_form/types.ts [hunk 2]
```diff
@@ -79,7 +79,7 @@ export interface WorkspaceFormProps {
   application: ApplicationStart;
   savedObjects: SavedObjectsStart;
   onSubmit?: (formData: WorkspaceFormSubmitData) => void;
-  defaultValues?: WorkspaceFormData;
+  defaultValues?: Partial<WorkspaceFormData>;
```
Comment (wanglam) [inline src/plugins/workspace/public/components/workspace_form/types.ts:82]: The `WorkspaceFormProps` was been used in `WorkspaceDetailedForm`, shall we separate a new `WorkspaceDetailedFormProps` like below code?
```ts
export interface WorkspaceDetailedFormProps extends WorkspaceFormProps {
  defaultValues?: WorkspaceFormData;
}
```
The `defaultValues` in `WorkspaceFormProps` should be `Partial<WorkspaceFormSubmitData>`.

File: src/plugins/workspace/public/components/workspace_form/use_workspace_form.ts [hunk 3]
```diff
@@ -80,6 +80,7 @@ export const useWorkspaceForm = ({
   const numberOfChanges = defaultValuesRef.current
     ? getNumberOfChanges(formData, {
         ...defaultValuesRef.current,
+        name: defaultValuesRef.current.name || '',
```
Comment (wanglam) [inline src/plugins/workspace/public/components/workspace_form/use_workspace_form.ts:83]: Shall we update the type definition of `getNumberOfChanges` to support partial form data? Then we don't need to fallback to an empty name.

Comment (Kapian1234) [reply src/plugins/workspace/public/components/workspace_form/use_workspace_form.ts:83]: Thanks for your suggestions!

File: src/plugins/workspace/public/components/workspace_creator/workspace_creator.tsx [hunk 4]
```diff
@@ -37,6 +44,11 @@ export const WorkspaceCreator = (props: WorkspaceCreatorProps) => {
     workspaceClient: WorkspaceClient;
     dataSourceManagement?: DataSourceManagementPluginSetup;
   }>();
+
+  const defaultWorkspaceFormValues: Partial<WorkspaceFormData> = {
```
Comment (SuZhou-Joe) [inline src/plugins/workspace/public/components/workspace_creator/workspace_creator.tsx:48]: Shouldn't we move the color stuff to the useWorkspaceForm hook?

File: src/plugins/workspace/public/components/workspace_form/utils.ts [hunk 5]
```diff
@@ -454,7 +454,7 @@ export const getNumberOfChanges = (
       Pick<WorkspacePermissionSetting, 'id'> & Partial<WorkspacePermissionSetting>
     >;
   },
-  initialFormData: Omit<WorkspaceFormData, 'id'>
+  initialFormData: Omit<Partial<WorkspaceFormData>, 'id'>
```
Comment (SuZhou-Joe) [inline src/plugins/workspace/public/components/workspace_form/utils.ts:457]:
```suggestion
  initialFormData: Partial<Omit<WorkspaceFormData>, 'id'>
```
Should like this?

File: src/plugins/workspace/public/components/workspace_creator/workspace_creator.tsx (hunk 4 above)
Comment (wanglam) [reply src/plugins/workspace/public/components/workspace_creator/workspace_creator.tsx:48]: Do you mean added it in the `useState` of color? I think the default value should be in workspace create page right?  User may confused if they didn't set the color but we display a default value in the workspace details page for legacy data.

Comment (SuZhou-Joe) [reply src/plugins/workspace/public/components/workspace_creator/workspace_creator.tsx:48]: I see, makes sense.

**Review by ruanyl — DISMISSED** [review]

File: src/plugins/workspace/public/components/workspace_form/utils.ts (hunk 5 above)
Comment (Kapian1234) [reply src/plugins/workspace/public/components/workspace_form/utils.ts]: Thank you! It makes it clearer.

**Review by ruanyl — APPROVED** [review]

**Review by SuZhou-Joe — APPROVED** [review]

**Review by wanglam — APPROVED** [review]

**Pull request merged**